src/main/java/de/tum/cit/aet/artemis/iris/service/pyris/PyrisWebhookService.java

```diff
@@ -240,4 +245,83 @@ public Map<Long, IngestionState> getLectureUnitsIngestionState(long courseId, lo
                 .collect(Collectors.toMap(DomainObject::getId, unit -> pyrisConnectorService.getLectureUnitIngestionState(courseId, lectureId, unit.getId())));
     }
 
+    private boolean faqIngestionEnabled(Course course) {
+        return irisSettingsService.getRawIrisSettingsFor(course).getIrisFaqIngestionSettings() != null
+                && irisSettingsService.getRawIrisSettingsFor(course).getIrisFaqIngestionSettings().isEnabled();
+    }
+
+    /**
+     * send the updated / created faqs to Pyris for ingestion if autoLecturesUpdate is enabled
+     *
+     * @param courseId Id of the course where the attachment is added
+     * @param newFaq   the new faqs to be sent to pyris for ingestion
+     */
+    public void autoUpdateFaqInPyris(Long courseId, Faq newFaq) {
+        IrisCourseSettings courseSettings = irisSettingsRepository.findCourseSettings(courseId).isPresent() ? irisSettingsRepository.findCourseSettings(courseId).get() : null;
+        if (courseSettings != null && courseSettings.getIrisFaqIngestionSettings() != null && courseSettings.getIrisFaqIngestionSettings().isEnabled()
+                && courseSettings.getIrisFaqIngestionSettings().getAutoIngestOnFaqCreation()) {
+            addFaqToPyris(newFaq);
+        }
+    }
+
+    /**
+     * adds the faq to the vector database in Pyris
+     *
+     * @param faq The faq that got Updated
+     * @return jobToken if the job was created else null
+     */
+    public String addFaqToPyris(Faq faq) {
+        if (faqIngestionEnabled(faq.getCourse())) {
+            return executeFaqAdditionWebhook(new PyrisFaqWebhookDTO(faq.getId(), faq.getQuestionTitle(), faq.getQuestionAnswer(), faq.getCourse().getId(),
+                    faq.getCourse().getTitle(), faq.getCourse().getDescription()));
+        }
+        return null;
+    }
+
+    /**
+     * executes the faq addition webhook to add faq to the vector database on pyris
+     *
+     * @param toUpdateFaq The faq that got Updated as webhook DTO
+     * @return jobToken if the job was created else null
+     */
+
+    private String executeFaqAdditionWebhook(PyrisFaqWebhookDTO toUpdateFaq) {
+        String jobToken = pyrisJobService.addFaqIngestionWebhookJob(toUpdateFaq.courseId(), toUpdateFaq.faqId());
+        PyrisPipelineExecutionSettingsDTO settingsDTO = new PyrisPipelineExecutionSettingsDTO(jobToken, List.of(), artemisBaseUrl);
+        PyrisWebhookFaqIngestionExecutionDTO executionDTO = new PyrisWebhookFaqIngestionExecutionDTO(toUpdateFaq, settingsDTO, List.of());
+        pyrisConnectorService.executeFaqAdditionWebhook(toUpdateFaq, executionDTO);
+        return jobToken;
+
+    }
+
+    /**
+     * delete the faqs from the vector database on pyris
+     *
+     * @param faq The faqs that got Updated / erased
+     * @return jobToken if the job was created
+     */
+    public String deleteFaqFromPyrisDB(Faq faq) {
```

Comment:
for instance same here: why is it relevant for Artemis that it's stored in a vector DB? Analogous, you wouldn't have an REST-endpoint in Artemis called "deleteProgrammingExerciseFromDB" - rather "deleteProgrammingExercise".